File: 秋招/lenovo/002/sol.js
const getSort = (nums, n) => {
  let res = [];
  for (let item of nums) {
    res.push(item.split(" "))
  }
  res.sort((a, b) => {
    return b[1] * 1 - a[1] * 1;
  })
  for (let i = 0; i < n; i++) {
    console.log(res[i][0]);
  }
}
let n = 4;
let nums = [
  'a 1',
  'b 2',
  'c 2',
  'd 1'
];
getSort(nums, n)

File: 回溯法/78_子集/sol.py
# class Solution:
#     def subsets(self, nums: List[int]) -> List[List[int]]:
#         res = []
#         for i in range(len(nums)+1):
#             for item in itertools.combinations(nums, i):
#                 res.append(item)
#         return res

from typing import List

# class Solution:
#     def subsets(self, nums: List[int]) -> List[List[int]]:
#         res = [[]]
#         for i in nums:
#             res = res + [[i] + num for num in res]
#             print(res)
#         return res


class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
      res = []
      n = len(nums)
      def backtracking(i, temp):
        res.append(temp)
        for j in range(i, n):
          backtracking(j+1, temp+[nums[j]])
      backtracking(0, [])
      return res

Sol = Solution()
res = Sol.subsets([1,2,3])
print(res)
